client2/src/shared/state/GUI.js

```diff
@@ -33,6 +35,14 @@ class GUI {
             this.cursorInTopSide = false;
         }
     }
+
+    setChatInputStatus(value) {
+        this.chatInputStatus = value;
+    }
+
+    getChatInputStatus() {
+        return this.chatInputStatus;
+    }
```

Comment:
Here's the setter and getter for chat input status. 

@Arcanorum May I know the reason why we do have setter but no getter methods in this file?

Reply:
Getter methods are pointless for just getting a property value like this, as
`getChatInputStatus();`
does the exact same thing as just
`GUIState.chatInputStatus;`

Using a method for this is only useful if you want to run side-effects when you get the value, but imo that is an anti-pattern. Side effects generally only happen when something changes (setter).

Reply:
I see so this is really intentional.

I really think that a getter (though somewhat redundant) will be better especially once you need to refactor or add a side effect in the future.

I intend to show you this, just to clarify the reason behind it.
Anyway, I'll revert the implementation to the one without getter accessor.

Reply:
Refactored at eebcc02